wagtail/wagtaildocs/views/documents.py

```diff
@@ -111,21 +116,26 @@ def edit(request, document_id):
         original_file = doc.file
         form = DocumentForm(request.POST, request.FILES, instance=doc)
         if form.is_valid():
-            if 'file' in form.changed_data:
-                # if providing a new document file, delete the old one.
-                # NB Doing this via original_file.delete() clears the file field,
-                # which definitely isn't what we want...
-                original_file.storage.delete(original_file.name)
-            doc = form.save()
-
-            # Reindex the document to make sure all tags are indexed
-            for backend in get_search_backends():
-                backend.add(doc)
-
-            messages.success(request, _("Document '{0}' updated").format(doc.title), buttons=[
-                messages.button(reverse('wagtaildocs:edit', args=(doc.id,)), _('Edit'))
-            ])
-            return redirect('wagtaildocs:index')
+            try:
+                if 'file' in form.changed_data:
+                    # if providing a new document file, delete the old one.
+                    # NB Doing this via original_file.delete() clears the file field,
+                    # which definitely isn't what we want...
+                    original_file.storage.delete(original_file.name)
+
+                with transaction.atomic():
+                    doc = form.save()
+            except OSError as err:
+                report_os_error(request, err)
```

Comment:
That's a fair point, actually! While it makes sense to report image upload errors back to the user in a friendly way (because otherwise they'll be buried by the AJAX call), I'm not sure that's the case for the main add/edit document form - arguably, if there's a legitimate server-level error then it's not really Wagtail's responsibility to 'prettify' it, and we should leave it to fail in a predictable plain-vanilla way (i.e. this PR should be closed as 'won't fix').